### PR TITLE
Use `RE2` for yang mac-address parsing

### PR DESCRIPTION
**Pull request by pudelkoM**
Addressed issues:

- `::google::protobuf::uint64` -> `uint64`
- `<regex>` -> RE2
- make `YangStringToMacAddress` safer
- comments
- style

This PR also fixes things introduces in commit https://github.com/stratum/stratum/commit/6e71349660bd1dbde65a9c5b765e391559e6a288.

### CODE REVIEW

File: stratum/hal/lib/common/yang_parse_tree_paths.cc [hunk 1]
```diff
@@ -1706,10 +1701,10 @@ void SetUpInterfacesInterfaceEthernetStateAutoNegotiate(
 // data from the DataResponse proto received from SwitchInterface, i.e.,
 // "&DataResponse::PortCounters::message", where message field in
 // DataResponse::Counters.
-TreeNodeEventHandler GetPollCounterFunctor(
-    uint64 node_id, uint32 port_id,
-    ::google::protobuf::uint64 (PortCounters::*func_ptr)() const,
-    YangParseTree* tree) {
+TreeNodeEventHandler GetPollCounterFunctor(uint64 node_id, uint32 port_id,
+                                           uint64 (PortCounters::*func_ptr)()
+                                               const,
```
Comment (Yi-Tseng) [inline stratum/hal/lib/common/yang_parse_tree_paths.cc:1706]: is there a better way to make `const` and the `func_ptr` in the same line?

File: stratum/hal/lib/common/yang_parse_tree_test.cc [hunk 2]
```diff
@@ -1161,13 +1161,14 @@ TEST_F(YangParseTreeTest,
                               /* Notification will not be called */ nullptr),
               StatusIs(_, _, ContainsRegex("not a TypedValue message")));
 
-  for (auto mac_string : kMacStrings) {
+  for (auto mac_string : kInvalidMacStrings) {
```
Comment (pudelkoM) [inline stratum/hal/lib/common/yang_parse_tree_test.cc:1164]:
```suggestion
  for (const auto& mac_string : kInvalidMacStrings) {
```

File: stratum/hal/lib/common/yang_parse_tree_paths.cc (hunk 1 above)
Comment (pudelkoM) [reply stratum/hal/lib/common/yang_parse_tree_paths.cc:1706]: Yes! I think the yang tree arg should come before the func_ptr, like with all the other `Get*Functor` functions.

File: stratum/hal/lib/common/utils.cc [hunk 3]
```diff
@@ -318,34 +316,33 @@ bool ConvertTrunkMemberBlockStateToBool(const TrunkMemberBlockState& state) {
   return state == TRUNK_MEMBER_BLOCK_STATE_FORWARDING;
 }
 
-std::string MacAddressToYangString(
-    const ::google::protobuf::uint64& mac_address) {
+std::string MacAddressToYangString(const uint64& mac_address) {
   return absl::StrFormat("%x:%x:%x:%x:%x:%x", (mac_address >> 40) & 0xFF,
```
Comment (bocon13) [inline stratum/hal/lib/common/utils.cc:320]:
```suggestion
  return absl::StrFormat("%2x:%2x:%2x:%2x:%2x:%2x", (mac_address >> 40) & 0xFF,
```

File: stratum/hal/lib/common/yang_parse_tree_test.cc [hunk 4]
```diff
@@ -1147,8 +1147,8 @@ TEST_F(YangParseTreeTest,
       "11:22:333:44:55:66",    // Too many hex digits
```
Comment (bocon13) [inline stratum/hal/lib/common/yang_parse_tree_test.cc:1147]:
```suggestion
      "0:0:0:0:0:0",         // Single digits not support
      "11:22:333:44:55:66",    // Too many hex digits
```

File: stratum/hal/lib/common/utils.cc (hunk 3 above)
Comment (pudelkoM) [reply stratum/hal/lib/common/utils.cc]: It was `%02x` for zero padding in the end. Feel free to extend the new tests cases.

File: stratum/hal/lib/common/utils.cc [hunk 5]
```diff
@@ -268,22 +267,21 @@ std::string ConvertSpeedBpsToString(
...
 }
 
-::google::protobuf::uint64 ConvertStringToSpeedBps(
-    const std::string& speed_string) {
-  if (speed_string.compare("SPEED_10GB") == 0) {
+uint64 ConvertStringToSpeedBps(const std::string& speed_string) {
+  if (speed_string == "SPEED_10GB") {
     return kTenGigBps;
-  } else if (speed_string.compare("SPEED_20GB") == 0) {
+  } else if (speed_string == "SPEED_20GB") {
```
Comment (pudelkoM) [inline stratum/hal/lib/common/utils.cc:273]: For reference: https://stackoverflow.com/a/9159670

File: stratum/hal/lib/common/utils.cc (hunk 3 above)
Comment (bocon13) [reply stratum/hal/lib/common/utils.cc]: Yes, sorry.

File: stratum/hal/lib/common/yang_parse_tree_paths.cc [hunk 6]
```diff
@@ -1343,7 +1339,7 @@ void SetUpInterfacesInterfaceEthernetConfigMacAddress(uint64 node_id,
       return status;
     }
```
Comment (pudelkoM) [inline stratum/hal/lib/common/yang_parse_tree_paths.cc:1340]: This feels like `RETURN_IF_ERROR`, but check other places.

**Pull request merged**